### ISSUE TITLE
Review #3

### ISSUE THREAD

**Issue by distillpub-reviewers** [distillpub/post--bayesian-optimization#11]
_The following peer review was solicited as part of the Distill review process._

_**The reviewer chose to waive anonymity.** Distill offers reviewers a choice between anonymous review and offering reviews under their name. Non-anonymous review allows reviewers to get credit for the service them offer to the community._

_Distill is grateful to **Jasper Snoek** for taking the time to review this article._

---

_Distill employs a [reviewer worksheet](https://docs.google.com/document/d/16BNLdSmoUc1zqydECJ8trYkVNDnKVlxT57EEXhCxWSc/edit) as a help for reviewers._

_The first three parts of this worksheet ask reviewers to rate a submission along certain dimensions on a scale from 1 to 5. While the scale meaning is consistently "higher is better", please [read the explanations for our expectations for each score](https://docs.google.com/document/d/16BNLdSmoUc1zqydECJ8trYkVNDnKVlxT57EEXhCxWSc/edit)—we do not expect even exceptionally good papers to receive a perfect score in every category, and expect most papers to be around a 3 in most categories._

_Any concerns or conflicts of interest that you are aware of?_: **No known conflicts of interest**
_What type of contributions does this article make?_: **This article gives an intuitive explanation of the basics of Bayesian optimization and it is an enjoyable and interesting read.  It does a nice job of visually demonstrating the impact and the behavior of a variety of choices made within Bayesian optimization.  It does a great job visualizing the various acquisition functions in Bayesian optimization with the help of nice interactive plots and animations.**

Advancing the Dialogue | Score
-- | --
How significant are these contributions? | 4/5

## Comments
I think this is a really clean and concise introduction to Bayesian optimization and some of the nuances of the underlying strategy followed.  Bayesian optimization is certainly dear to me and I appreciate having someone take the time to produce nice visualizations so I feel inclined to accept.  There is a lot that could be added to this post, but I suppose in the spirit of the journal (i.e. short and crisp) this might be just right?  In any case there are some underlying modeling issues that I would like corrected before this is accepted.

Outstanding Communication | Score
-- | --
Article Structure | 4/5
Writing Style | 5/5
Diagram & Interface Style | 4/5
Impact of diagrams / interfaces / tools for thought? | 3/5
Readability | 4/5

## Comments
- I think the article is structured well and flows nicely.
- Writing style is very easy to follow and accessible.  A couple of minor typos.
- The diagrams are intuitive and I love having interactive plots of acquisition functions in Bayesian optimization.
- The diagrams are really helpful, elegant and really drive the story.  However, they aren’t terribly novel since this is exactly how we’ve been visualizing Bayesian optimization in papers and texts for years.  The authors do turn them into animations, however, which is a neat upgrade from the static plots previously used.
- Very readable.  The authors side-step some of the more technical aspects of Bayesian modeling (e.g. how does a Gaussian process work and how do you fit it).  However, for the purposes of this article that might be better.

Scientific Correctness & Integrity | Score
-- | --
Are claims in the article well supported? | 4/5
Does the article critically evaluate its limitations? How easily would a lay person understand them? | 4/5
How easy would it be to replicate (or falsify) the results? | 5/5
Does the article cite relevant work? | 4/5
Does the article exhibit strong intellectual honesty and scientific hygiene? | 4/5

## Comments
- I think there’s a lot of specifics, challenges and follow up work that isn’t referenced here.  However, the authors do a good job of citing relevant work.
- This is just an introductory exposition that would be really easy to replicate empirically.
- Yes. There is a lot of relevant work missing, e.g. citations for Thompson sampling (http://proceedings.mlr.press/v84/kandasamy18a/kandasamy18a.pdf) but most of the relevant citations are there.  
- There is what I see as a critical modeling mistake in the diagrams and resulting analysis. The authors don’t adjust for the mean of the data, which results in weird and pathological behavior.  I would like the authors to correct this and then resubmit (and then I think it should be ok).  Detailed comments are below.

## Detailed comments
Intro:
Note the word “hyperparameter” can actually be contentious.  The traditional definition of a hyperparameter is as a higher level model parameter influencing the parameters of the model.  Under that definition things like learning rate, optimization parameters, etc. don’t really apply.  So I always say Bayesian optimization is used to tune hyperparameters, optimization parameters and other model parameters.

Mining Gold:
There’s a neat historical precedent to this that’s worth mentioning.  The first use of Gaussian processes was actually to model ore density in South Africa.  Applying Gaussian processes was initially called “Kriging” after Danie Krige (https://en.wikipedia.org/wiki/Danie_G._Krige) who used GPs to model the spatial density of ore deposits (and figure out where to drill).

“Active Learning”:
“We cannot estimate the gold estimate” sounds awkward.  Maybe rephrase.

“Gaussian processes”:
“Gaussian processes regression” -> Gaussian process regression
“(Smoothess) Such” -> “(Smoothness). Such”

“Prior model”:
The Matern kernel is a specific choice of prior that is worth spending some time rationalizing.  It’s probably a good idea to introduce the concept of a kernel and how the choice of kernel corresponds to a prior over functions.  Then describe how the Matern lets you determine the smoothness of the prior (i.e. a Matern 5/2 means twice differentiable).  How does that correspond to your assumptions about gold smoothness?

“UCB”:
“This is because while the variance or uncertainty is high for such points, the posterior mean is low.” 
The posterior mean is low because the prior mean is set to 0.  You could subtract the mean from the observed data to make it 0 mean (then in your case the mean would go to ~5 instead of 0 and the acquisition function would be higher further from the data).  It would be better to make the mean a hyperparameter of the GP and optimize it or integrate it out.  

“Probability of Improvement”:
“given same exploitability” -> “given the same...”

The “hero plot” is neat.  Though I’m not sure I understand why it’s called a hero plot.  

“(can be identified by the grey translucent area” is missing closing parens. 

“Expected Improvement”:
These plots are neat and convey the intuition really nicely.  However, the EI values are tiny and the behavior does not follow what I have seen for EI.  I suspect the optimization routine is under exploring because of the zero-mean issue I brought up above.  Specifically, for a stationary kernel, the GP posterior will return to the mean when moving away from the data.  In this case, it’s returning to 0, which is silly since 0 is not the mean of the observations you have seen.  I suspect the routine will be much better behaved if you subtract out the mean of the data, fit the model, and then add the mean back in.

“Gaussian Process Upper Confidence Bound (GP-UCB)”:
I like the discussion of regret.  However, I don’t think Srinivas et al. introduced GP-UCB and other acquisition functions also minimize regret.  Instead they derived some elegant bounds on regret under the GP-UCB acquisition function.  I think I would rephrase this to say something like: “Srinivas et al. developed a schedule for \beta that they theoretically demonstrated minimizes cumulative regret.  The schedule is T_t … “
